packages/data-connect/test/unit/fetch.test.ts

```diff
@@ -74,8 +80,142 @@ describe('fetch', () => {
+        await dcFetch(
+          'http://localhost',
+          {
+            name: 'n',
+            operationName: 'n',
+            variables: {}
+          },
+          {} as AbortController,
+          null,
+          null,
+          null,
+          false, // _isUsingGen is false
+          callerSdkType as CallerSdkType
+        );
+      }
+    }
+
+    // an example of the args to fetch():
+    // [
+    //    "http://localhost",
+    //    {
+    //      "body": "{\"name\":\"n\",\"operationName\":\"n\",\"variables\":{}}",
+    //      "headers": {
+    //        "Content-Type": "application/json",
+    //        "X-Goog-Api-Client": "gl-js/ fire/11.2.0"
+    //      },
+    //      "method": "POST",
+    //      "signal": [undefined]
+    //    }
+    // ]
+    for (let i = 0; i < fakeFetchImpl.args.length; i++) {
+      const args = fakeFetchImpl.args[i];
+      expect(args.length).to.equal(2);
+      expect(Object.prototype.hasOwnProperty.call(args[1], 'headers')).to.be
+        .true;
+      expect(
+        Object.prototype.hasOwnProperty.call(
+          args[1]['headers'],
+          'X-Goog-Api-Client'
+        )
+      ).to.be.true;
+      expect(typeof args[1]['headers']['X-Goog-Api-Client']).to.equal('string');
+
+      const xGoogApiClientValue: string =
+        args[1]['headers']['X-Goog-Api-Client'];
+      // the sdk type headers are always of the form "js/xxx", where xxx is _callerSdkType.toLower()
+      // when the _callerSdkType is Base, we do not set any header
+      // when the _callerSdkType is Generated, we use "js/gen" instead of "js/generated"
+      if (callerSdkTypesUsed[i] === CallerSdkTypeEnum.Base) {
+        expect(xGoogApiClientValue).to.not.match(RegExp(`js\/w`));
+      } else if (callerSdkTypesUsed[i] === CallerSdkTypeEnum.Generated) {
+        expect(xGoogApiClientValue).to.match(RegExp(`js\/gen`));
+      } else {
+        expect(xGoogApiClientValue).to.match(
+          RegExp(`js\/${callerSdkTypesUsed[i].toLowerCase()}`)
+        );
+      }
+    }
+  });
+  it('should assign custom headers based on _callerSdkType before checking to-be-deprecated _isUsingGen', async () => {
+    const json = {
+      code: 200,
+      message1: 'success'
+    };
+    const fakeFetchImpl = mockFetch(json, false);
+
+    const callerSdkTypesUsed: string[] = [];
+
+    for (const callerSdkType in CallerSdkTypeEnum) {
+      if (
+        Object.prototype.hasOwnProperty.call(CallerSdkTypeEnum, callerSdkType)
+      ) {
+        callerSdkTypesUsed.push(callerSdkType);
+        await dcFetch(
+          'http://localhost',
+          {
+            name: 'n',
+            operationName: 'n',
+            variables: {}
+          },
+          {} as AbortController,
+          null,
+          null,
+          null,
+          true, // _isUsingGen is true
+          callerSdkType as CallerSdkType
+        );
+      }
+    }
+
+    for (let i = 0; i < fakeFetchImpl.args.length; i++) {
```

Comment:
Same comment here. We should make the check right after making the call to `dcFetch`